File: Data Structures & Algorithms/sliding-window-maximum/submission-0.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        int left = 0;
        int right = k - 1;
        vector<int> ans;
        while(right < nums.size()){
            int max = *max_element(nums.begin() + left, nums.begin() + right+1);
            ans.push_back(max);
            left++;
            right++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.max_diameter = 0

        def dfs(node):
            if not node:
                return 0
            #left + right is the height of the present node
            left = dfs(node.left) 
            right = dfs(node.right)
            self.max_diameter = max(self.max_diameter, left + right)
            return 1 + max(left, right) # 1 + height of children

        dfs(root)
        return self.max_diameter